drizzlepac/hlautils/se_source_generation.py

```diff
@@ -0,0 +1,455 @@
+#!/usr/bin/env python
+"""Utility to create the Sextractor-like source catalog using PhotUtils
+
+The function, create_sextractor_like_sourcelists, processes the provided
+input file (presumptively a drizzled, white-light image) looking for sources.
+A segmentation map and the total detection product catalog are the critical 
+outputs of this routine.
+
+The function, measure_source_properties, uses the segmentation map generated
+by create_sextractor_like_sourcelists, and tries to measure various properties
+of the objects at the location of the found sources.  The measurements are
+done in the drizzled, specific filter images of the visit. The filter product
+is the output of the routine.
+
+"""
+import sys
+from distutils.version import LooseVersion
+
+import numpy as np
+try:
+    from matplotlib import pyplot as plt
+except Exception:
+    plt = None
+
+import astropy.units as u
+from astropy.io import ascii
+from astropy.io import fits as fits
+from astropy.table import Column
+from astropy.convolution import Gaussian2DKernel, MexicanHat2DKernel
+from astropy.stats import gaussian_fwhm_to_sigma
+import photutils
+from photutils import detect_sources, source_properties, deblend_sources
+from photutils import Background2D, MedianBackground, SExtractorBackground, StdBackgroundRMS
+from stwcs.wcsutil import HSTWCS
+from stsci.tools import logutil
+
+
+__taskname__ = "se_source_generation"
+
+log = logutil.create_logger(__name__, level=logutil.logging.INFO, stream=sys.stdout)
+
+_all__ = ["create_sextractor_like_sourcelists", "measure_source_properties"]
+
+def create_sextractor_like_sourcelists(source_filename, catalog_filename, param_dict, se_debug=False):
+    """Use photutils to find sources in image based on segmentation.
+
+    Parameters
+    ----------
+    source_filename : string
+        Filename of the "white light" drizzled image (aka the total detection product) which
+        is used for the detection of sources
+
+    catalog_filename : string
+        Name of the output source catalog for the total detection product
+
+    param_dict : dictionary
+        dictionary of drizzle, source finding, and photometric parameters
+
+    se_debug : bool, optional
+        Specify whether or not to plot the image and segmentation image for
+        visualization and debugging purposes
+
+    Returns
+    -------
+    segm : `photutils.segmentation.SegmentationImage`
+        Two-dimensional segmentation image where found source regions are labeled with
+        unique, non-zero positive integers.
+
+    kernel :
+
+    bkg : `~photutils.background.Background2D` or None
+        A background map based upon the `~photutils.background.SExtractorBackground`
+        estimator
+
+    bkg_rms_mean : float
+        Mean bkg.background FIX
+
+    """
+
+    # Open the "white light" image and get the image data
+    imghdu = fits.open(source_filename)
```

Comment:
It may be helpful to support both a filename or an already opened FITS HDUList object as input here.  That would allow this code to be called with an in-memory copy of the images instead of forcing the calling code to write it to disk before calling this code.  This is NOT something that needs to be done now, but rather is a refinement for future development.